src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -1,67 +1,212 @@
+
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
 
-class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig)(implicit val system: ActorSystem, val executionContext: ExecutionContext)
+class GoogleDataprocDAO(protected val dc: DataprocConfig)(implicit val system: ActorSystem, val executionContext: ExecutionContext)
   extends DataprocDAO with GoogleUtilities {
 
   private val httpTransport = GoogleNetHttpTransport.newTrustedTransport
   private val jsonFactory = JacksonFactory.getDefaultInstance
-  private val cloudPlatformScopes = List(PubsubScopes.CLOUD_PLATFORM)
+  private val cloudPlatformScopes = List(ComputeScopes.CLOUD_PLATFORM)
+  private val storageScopes = Seq(StorageScopes.DEVSTORAGE_FULL_CONTROL, ComputeScopes.COMPUTE, PlusScopes.USERINFO_EMAIL, PlusScopes.USERINFO_PROFILE)
+  private val vmScopes = List(ComputeScopes.COMPUTE, ComputeScopes.CLOUD_PLATFORM)
+  private val serviceAccountPemFile = new File(dc.configFolderPath, dc.serviceAccountPemName)
 
   private def getDataProcServiceAccountCredential: Credential = {
     new GoogleCredential.Builder()
       .setTransport(httpTransport)
       .setJsonFactory(jsonFactory)
-      .setServiceAccountId(dataprocConfig.serviceAccount)
+      .setServiceAccountId(dc.serviceAccount)
       .setServiceAccountScopes(cloudPlatformScopes.asJava)
-      .setServiceAccountPrivateKeyFromPemFile(new java.io.File(dataprocConfig.serviceAccountPemPath))
+      .setServiceAccountPrivateKeyFromPemFile(serviceAccountPemFile)
+      .build()
+  }
+
+  def getBucketServiceAccountCredential: Credential = {
+    new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(dc.serviceAccount)
+      .setServiceAccountScopes(storageScopes.asJava) // grant bucket-creation powers
+      .setServiceAccountPrivateKeyFromPemFile(serviceAccountPemFile)
       .build()
   }
 
+  private def getVmServiceAccountCredential: Credential = {
+    new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(dc.serviceAccount)
+      .setServiceAccountScopes(vmScopes.asJava)
+      .setServiceAccountPrivateKeyFromPemFile(serviceAccountPemFile)
+      .build()
+  }
 
   def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[ClusterResponse] = {
-    val op = build(googleProject, clusterName, clusterRequest)
-    op.map{op =>
-      val metadata = op.getMetadata
-      ClusterResponse(clusterName, googleProject, metadata.get("clusterUuid").toString, metadata.get("status").toString, metadata.get("description").toString, op.getName)}
+    val bucketName = s"${clusterName}-${UUID.randomUUID.toString}"
+
+    val bucketResponse = initializeBucket(googleProject, clusterName, bucketName, clusterRequest.serviceAccount)
```

Comment:
the bucket should live in leo's google project and be private to leo, not the user's